apps/arweave/src/ar_nonce_limiter_client.erl

```diff
@@ -0,0 +1,157 @@
+-module(ar_nonce_limiter_client).
+
+-behaviour(gen_server).
+
+-export([start_link/0]).
+
+-export([init/1, handle_cast/2, handle_call/3, handle_info/2, terminate/2]).
+
+-include_lib("arweave/include/ar.hrl").
+-include_lib("arweave/include/ar_config.hrl").
+
+-record(state, {
+	remote_servers
+}).
+
+%%%===================================================================
+%%% Public interface.
+%%%===================================================================
+
+%% @doc Start the server.
+start_link() ->
+	gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+%%%===================================================================
+%%% Generic server callbacks.
+%%%===================================================================
+
+init([]) ->
+	process_flag(trap_exit, true),
+	case ar_config:use_remote_vdf_server() andalso ar_config:pull_from_remote_vdf_server() of
+		false ->
+			ok;
+		true ->
+			gen_server:cast(?MODULE, pull)
+	end,
+	{ok, Config} = application:get_env(arweave, config),
+	RawPeers = queue:from_list(Config#config.nonce_limiter_server_trusted_peers),
+	{ok, #state{ remote_servers = RawPeers }}.
+
+handle_call(Request, _From, State) ->
+	?LOG_WARNING("event: unhandled_call, request: ~p", [Request]),
+	{reply, ok, State}.
+
+handle_cast(pull, State) ->
+	ar_util:cast_after(800, ?MODULE, pull),
```

Comment:
I'm not sure how much it matters, but with a fixed polling rate, the client will likely get some VDF steps pretty "late". 
i.e. if the server is generating a new step every second, the client will get:
- step 1, at 1.6 seconds (0.6 seconds late)
- step 2, at 2.4 seconds (0.4 seconds late)
- step 3, at 3.2 seconds (0.2 seconds late)
- step 4, at 4 seconds (ontime)
- step 5 at 5.6 seconds (0.6 seconds late)
- etc...

However since the lateness doesn't compound I'm not sure how material this is for miners

One fix might be to have the server include information in its response about when to next pull (e.g. based on an average of how fast its computing VDF and when it last computed the VDF)